connectivity/check/logging.go

```diff
@@ -345,11 +345,12 @@ func (a *Action) Infof(format string, s ...interface{}) {
 }
 
 // Warn must be called when a warning is detected performing the Action.
-//
+// Warnings are logged both inline and at the end of a test.
```

Comment:
Could we keep the `Warn*()` family of functions as-is? See my comment above, unexpected stderr is not a warning, it's a failure.

Reply:
@ti-mo I find it odd that one logging level is not logged inline while every other level is. Maybe my bad that I'm thinking of these logging functions in terms of "loglevels". I have nothing against highlighting specific log levels separately, but I think all logs should appear in the output in the order in which they were generated.

If we need warning functions that do not log inline we should call them something else instead to not cause confusion with usage of `Warn()` and `Warnf()` from other Go packages.

Reply:
Sure, I can see how echoing them inline in the test's log would be useful as well, but we need to keep in mind they are also repeated at the end of the test run. They're not supposed to be used frequently and we should avoid using them for multi-line output.

Or maybe we should just remove `Warn()` altogether. :) Being indecisive about events ("it's a failure but not really") is a slippery slope because that's how things start rotting. It was originally intended to be used for flagging deprecated things etc., but currently it's only used for the `Unable to contact Hubble Relay...` warning. If it causes more trouble/confusion than it's worth, let's remove it.

Reply:
I'll separate this change to a PR of it's own.